CDCgov/prime-reportstream
# 10491: sync schemas workflow

This PR adds the workflow for syncing translation schemas between different azure steps.

Test Steps:
There is pretty comprehensive testing for this, so the easiest way to test is to uncomment `SyncTranslationSchemasTest` the code that has it run only during CI and run the test

Manual steps
1. Upload a valid and previous valid blob into the azurite-stage metadata container
2. Upload sample schemas to the azurite-stage container
3. Upload a valid and previous valid blob into the azurite metadata container
4. Run the new sync schemas command with `metadata` as the container, azurite-stage as the source and azurite as the destination
5. Wait for the validation process to complete
6. Verify that azurite contains the schema

## Changes
- A command that syncs schemas between environments
- Azure functions triggered by a validating.txt file being present that validates the schemas and rollback if an error is encountered

## Checklist

### Testing

### Process

## Linked Issues
- #10491 
- #10494

## To Be Done

## Specific Security-related subjects a reviewer should pay specific attention to
